### ISSUE TITLE
HTTPError: 503 Server Error: Service Unavailable: Back-end server is at capacity for url: https://courses.ed...

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#2168]
@giocalitri  what do you expect happens when edX returns a 503 for the enrollment API like this? It doesn't happen that frequently, but I'd like to get an idea what the user impact is so we can decide how to prioritize it. 

https://sentry.io/mit-office-of-digital-learning/micromasters/issues/193330670/

```
HTTPError: 503 Server Error: Service Unavailable: Back-end server is at capacity for url: https://courses.edx.org/api/enrollment/v1/enrollment
(12 additional frame(s) were not displayed)
...
  File "rest_framework/views.py", line 474, in dispatch
    response = handler(request, *args, **kwargs)
  File "dashboard/views.py", line 57, in get
    return Response(get_user_program_info(request.user, edx_client))
  File "dashboard/api.py", line 124, in get_user_program_info
    CachedEdxDataApi.update_cache_if_expired(user, edx_client, cache_type)
  File "dashboard/api_edx_cache.py", line 282, in update_cache_if_expired
    update_func(user, edx_client)
  File "dashboard/api_edx_cache.py", line 165, in update_cached_enrollments
    enrollments = edx_client.enrollments.get_student_enrollments()

HTTPError: 503 Server Error: Service Unavailable: Back-end server is at capacity for url: https://courses.edx.org/api/enrollment/v1/enrollment
```

**Comment by giocalitri** [comment]
I am not sure what can we do here other than asking edX what is the maximum rate of requests they will serve for us: I think this is related to a general optimization that we need to start looking at.

**Comment by pdpinch** [comment]
Do we try again at a later time, or do we give up? Does this block the batch, or just the one request?

**Comment by noisecapella** [comment]
It only blocks the one request. The user can refresh and it makes another attempt

**Comment by giocalitri** [comment]
that one happened in the dashboard: I think we should think about returned the cashed data but tell the user that there was a problem with the backend and the data might be out of date

**Comment by giocalitri** [comment]
but this is a pretty big effort

**Comment by pdpinch** [comment]
Closed in favor of #2181
